### ISSUE TITLE
Accidental MXML Recursion Kills Entire Environment

### ISSUE THREAD

**Issue by brianraymes** [apache/royale-compiler#187]
I have stumbled across a doozy. Copying a file with the intent to rename it killed my entire environment and silently kills the ability to compile with no warnings or errors.

**Steps to Reproduce**

1. Have an MXML file, say TabbedContainer.mxml
2. Have it's root tag be the same as it's name:

```
<?xml version="1.0" encoding="utf-8"?>
<containers:TabbedContainer xmlns:fx="http://ns.adobe.com/mxml/2009"
    xmlns:j="library://ns.apache.org/royale/jewel"
    xmlns:js="library://ns.apache.org/royale/basic"
    xmlns:html="library://ns.apache.org/royale/html"
    xmlns:containers="com.some.path.to.containers.*">
    .  .  .
```

**Expected Results**

Editor should flag this as an error/issue. Compilation should fail with an error as this is obviously incorrect.

**Actual Results**

`java` processes lock up, `vscode-as3mxml` language server dies and fails to start, and all other functionality provided through it fails. Java processes kicked off from VS Code never exit and require one to forcefully end the tasks (exiting vs-code does not kill the java process that it spawned). Compilation fails by ever ending with no errors or warnings. I did not wait to see how long it actually stays in that "infinite compiling" state.

**Workaround**

None that I know of at this time.

**Comments**

This was an easy mistake to make, simply copied a file and renamed it and all went south from there. I spent a considerable amount of time looking into java versions, vs-code output, maven, rebooting, windows/java/vs-code/extension updates, etc. All because of a quick file copy/rename.

**Comment by joshtynjala** [comment]
Confirmed bug. Creating a file named _Application.mxml_ with the following content will cause the process to hang when trying to compile.

```
<js:Application xmlns:fx="http://ns.adobe.com/mxml/2009"
	xmlns:js="library://ns.apache.org/royale/basic">
</js:Application>
```

In vscode-as3mxml, code intelligence will not initialize if opening an existing workspace with this file. When adding the file to an open workspace, where the code intelligence was working correctly, it will suddenly stop working because the language server process (which uses the compiler) hangs.
